### ISSUE TITLE
`reportUnusedParameter` false negative

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#653]
```py
def f(a: int):  # no nerror
   a = 1
   print(a)
```

https://basedpyright.com/?typeCheckingMode=all&code=CYUwZgBGAUCGBcUA2B7WAXAlPAUBfEsEAvBAIx4EAOATgJYB26cmQA

**Comment by DetachHead** [comment]
huh???? it's literally used

**Comment by KotlinIsland** [comment]
fail, it's reassigned first, the passed value is never used

**Comment by DetachHead** [comment]
oh, that's the same with unused variables too. https://github.com/microsoft/pyright/issues/8806
